omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportDefinitionsWithScheduledRequestsResource.java

```diff
@@ -0,0 +1,146 @@
+
+  @Override
+  public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+    final DelegatingResourceDescription description = new DelegatingResourceDescription();
+    description.addProperty("reportDefinition", Representation.FULL);
+    description.addProperty("reportScheduleRequest", Representation.FULL);
+    return description;
+  }
+
+  @Override
+  public SimpleObject newDelegate() {
+    return new SimpleObject();
+  }
+
+  @Override
+  public SimpleObject save(SimpleObject scheduledReport) {
+    throw new ResourceDoesNotSupportOperationException();
+  }
+
+  @Override
+  public void purge(SimpleObject scheduledReport, RequestContext requestContext) throws ResponseException {
+    throw new ResourceDoesNotSupportOperationException();
+  }
+
+  @Override
+  protected void delete(SimpleObject scheduledReport, String s, RequestContext requestContext) throws ResponseException {
+    throw new ResourceDoesNotSupportOperationException();
+  }
+
+  @Override
+  public SimpleObject getByUniqueId(String s) {
+    throw new ResourceDoesNotSupportOperationException();
+  }
+
+  @Override
+  protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+    final List<ReportDefinition> reportDefinitions = getSortedReportDefinitions(context);
+    List<SimpleObject> scheduledReports = mergeScheduledReports(reportDefinitions);
+
+    return new AlreadyPaged<SimpleObject>(context, scheduledReports, false);
+  }
+
+  @Override
+  protected PageableResult doSearch(RequestContext context) {
+    return doGetAll(context);
+  }
+
+  private List<ReportDefinition> getSortedReportDefinitions(RequestContext context) {
+    final List<ReportDefinition> reportDefinitions =
+        Context.getService(ReportDefinitionService.class).getAllDefinitions(false);
+
+    Collections.sort(reportDefinitions, getComparatorForRequestSorting(context));
+    return reportDefinitions;
+  }
+
+  private List<SimpleObject> mergeScheduledReports(List<ReportDefinition> reportDefinitions) {
+    List<SimpleObject> scheduledReports = new ArrayList<SimpleObject>();
+
+    for (ReportDefinition reportDefinition : reportDefinitions) {
+      List<ReportRequest> scheduledReportRequests = Context
+          .getService(ReportService.class)
+          .getReportRequests(reportDefinition, null, null, ReportRequest.Status.SCHEDULED,
+              ReportRequest.Status.SCHEDULE_COMPLETED);
+
+      if (scheduledReportRequests.isEmpty()) {
+        scheduledReports.add(new SimpleObject()
+            .add("reportDefinition", reportDefinition)
+            .add("reportScheduleRequest", null));
+      } else {
+        for (ReportRequest reportRequest : scheduledReportRequests) {
+          scheduledReports.add(new SimpleObject()
+              .add("reportDefinition", reportDefinition)
+              .add("reportScheduleRequest", reportRequest));
+        }
```

Comment:
This is a very odd implementation.  Why wouldn't we just have an object for each report defintition like:
```
{
  reportDefinition: { },
  scheduledRequests: [ ] 
}
```

Where the scheduledRequests array would contain all of the report requests listed above, or would be empty if there are none...

Reply:
@mseaton indeed, it can be simplified